src/Plugin/RulesDataMatcher/ArrayContainsMatcher.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesDataMatcher\ArrayContainsMatcher.
+ */
+
+namespace Drupal\rules\Plugin\RulesDataMatcher;
+
+use Drupal\rules\Context\DataMatcherInterface;
+
+/**
+ * Defines an array contains matcher.
+ *
+ * @RulesDataMatcher(
+ *   id = "rules_array_contains",
+ *   label = @Translation("An array contains matcher."),
+ *   subject_type = "array",
+ *   object_type = "array"
```

Comment:
is not an array, but any value. thus should be "any". generally, it should be match the data type of the list - not sure how we'd specify that - but for now it should be enough to go with list and any.